### ISSUE TITLE
Produce bower-friendly individual module builds

### ISSUE THREAD

**Issue by jkymarsh** [angular-ui/bootstrap#2196]
Currently, pulling down `angular-bootstrap` via `bower` only provides the built lib for the entirety of angular-bootstrap; if you only need one or two modules, this is overkill. The [bower repo](https://github.com/angular-ui/bootstrap-bower) also doesn't contain individual modules, such as those produced by the "Create a Build" dialog on the website.

It'd be nice if the bower repo contained the built lib for the entirety of angular-bootstrap, as WELL as individual built libs for each module, much like [jQueryUI does](https://github.com/jquery/jquery-ui/tree/master/ui).

Hell, it wouldn't even need to be an official bower repo; as long as there were a repo somewhere that allowed you to pull in an individual module, I'd be happy. As it is, you HAVE to build the project to produce usable individual modules.

**Comment by kencaron** [comment]
+1

**Comment by AshCoolman** [comment]
+1

**Comment by HeberLZ** [comment]
This would greatly help improve download time!

**Comment by mlaccetti** [comment]
This ticket should be closed - the `bootstrap-bower` repo contains bower-friendly builds by version; no more `gh-pages` tag hack required.

**Comment by wesleycho** [comment]
This is currently fixed, as pointed out by @mlaccetti.

**Comment by chuckd** [comment]
@mlaccetti @wesleycho I can't see why this has been closed or how it has been fixed. I may just have missed something! I checked the latest angular-bootstrap in bower and it still contains only the builds with the whole library:

```bash
$ ls bower_components/angular-bootstrap
README.md
bower.json
index.js
package.json
ui-bootstrap-csp.css
ui-bootstrap-tpls.js
ui-bootstrap-tpls.min.js
ui-bootstrap.js
ui-bootstrap.min.js
```

The point of this issue is to request a separate .js file for each module (e.g. tooltip.{tpl.}js, typeahead.{tpl.}js). AngularStrap, as pointed out above, includes these individual files in the modules/ folder in its bower repo, so you can just pull in the bits you need.

**Comment by wesleycho** [comment]
Hmm, not sure what I was thinking when I closed this. But it should be noted that you can consume directly from `angular-ui-bootstrap` from bower and pull in the specific scripts you need from there.

**Comment by pezillionaire** [comment]
+1

**Comment by icfantv** [comment]
@pezillionaire please do not just +1 issues.  It adds no inherent value to the issue and clutters up the thread.

**Comment by pezillionaire** [comment]
@icfantv Fair enough, but how do I tell you or the keepers of this repo that I also would like this feature without cluttering up the thread? Does +1 not succinctly state that I, in addition to the many others above, would like to see this feature added to UI-Bootstrap? Does my reiteration of this issue, or what others have said before me really help when all I want/need to say is, "Yes, I too would like this."?

Yes, I would also like to have bower-friendly individual module builds of UI-Bootstrap so I don't have to weight down my project with 40 components I'm not using. I think this would be a great benefit to myself and other developers.

Thank you for making me feel welcome on my first post in your community.

**Comment by OldSneerJaw** [comment]
+1

**Comment by wesleycho** [comment]
I come back to every issue - I probably read all of the issues each month (sometimes multiple times), so they don't get forgotten.

One solution now is to point Bower to this repository and to take the scripts, build the templates of interest, and to use build tooling to combine it into your dependencies - it is not the simplest solution, but any change with the distributed build has a high overhead currently, especially since we are going to be strained for resources very soon on rapid development for an Angular 2 version with a hard deadline to have a preview ready.

**Comment by pkozlowski-opensource** [comment]
Few remarks:
* bower is going away so probably there is no point in investing in automating push of individual directives into it - especially that it would require creation of many git repositories  
* npm seems to be the way to go so we could probably publish individual directives in there. Still we need to think of / solve several pbs:
    * how to handle dependencies? (deps on repos vs. bundling deps in files)
    * what to publish (scripts + templates combined or separately? or both?)
    * ....

So it is not like we can't do but, as @wesleycho points out, this needs work so someone needs to step in or ... wait till one of the core contributors have the same use-case....

**Comment by icfantv** [comment]
@pezillionaire, That was a very good and thorough explanation and why I'd said "please do not *just* +1 issues."  Had you posted that explanation either along with or in lieu of your +1, that would have added value.

As far as your sarcastic closing comment goes, I apologize if you were offended.  I ask you to please take a step back and try to look at managing a popular OSS project through a project developer's eyes.  We get around a dozen issues every day, most of which are not bugs and are support requests or developers not taking the time to properly debug their code.  We've addressed how to property ask for and receive support on our website but most people don't bother to take the time to read it.  As you can imagine, that can get quite frustrating because it happens over and over again with seemingly no end.  OSS is very, very hard and most of us do it in our spare time, not as part of our full-time job.  The lucky few work for companies that allow time to work on OSS projects.

I view our job as not only developers and maintainers of an OSS project, but also as teachers.  If we can teach developers to "help us help you" then these things don't happen, time isn't wasted, and the entire community prospers.

**Comment by pezillionaire** [comment]
@wesleycho @pkozlowski-opensource and @icfantv Thank you all for your replies.

I appreciate what you are doing, especially as I do not have time to contribute to OSS outside of work.

@wesleycho Thanks! Right now, we are just adding the component, labeled as external, to our repo, it's a bit of saticfice scenario. Good luck with the Angular 2 release, looking forward to it!

@pkozlowski-opensource Was not aware of Bower's imminent demise. Is there something that will be taking over the role of managing app dependencies? npm? That being the case, I can understand the reluctance to spending time on this.

@icfantv Thank you for the apology. However in this case it seems my +1ing has stimulated discussion in an otherwise dormant thread. In the future, please don't just dismiss people who want to add a small amount of weight to an issue, it really felt disrespectful especially since I was not the first to do so in this thread.

**Comment by icfantv** [comment]
@pezillionaire, I think you missed my point.  Simply adding the text `+1` does not add any value whatsoever to an issue.  If one does it to show support and that they want the issue addressed, then one can add a thoughtful and well-reasoned comment indicating thus and, even better, submit a PR with the proposed fix after a discussion has been had as to the proper course of action, if any.

Your original comment was 'dismissed' because it, despite what you might think, didn't add any value and when pointed out, you took offense.

The irony, of course, is that in an effort prevent thread clutter with unproductive comments, this discourse has done the exact opposite.

So bottom line, in the future, please do not just add `+1` text to issues.

**Comment by wesleycho** [comment]
Closing this in favor of #4739 - this should come for free if we solve this there, and any decision will appear in that issue.
